src/extension/PostProblems.ts

```diff
@@ -0,0 +1,332 @@
+         * vscode handles this range by emphasizing the words that start at position [position]
+         */
+        const diagnostic: Diagnostic = this.createDiagnostic(
+          position,
+          position,
+          descriptiveMessage,
+        )
+        this.setDiagnosticMap(diagnostic, path, diagnosticMap)
+      }
+    }
+    this.postDiagnostics(diagnosticMap)
+  }
+
+  /**
+   * returns a new diagnostic object
+   * @param startPosition position where the problem starts
+   * @param endPosition position where the problem ends
+   * @param message message describing the problem
+   * @returns diagnostic object
+   */
+  private createDiagnostic(
+    startPosition: Position,
+    endPosition: Position,
+    message: string,
+  ): Diagnostic {
+    const range: Range = new Range(startPosition, endPosition)
+    const diagnostic = new Diagnostic(range, message)
+    return diagnostic
+  }
+
+  /**
+   * sets the diagnostic map [diagnosticMap] with the new value:
+   * @param diagnostic Diagnostic object
+   * @param path path to the file where problems originated from
+   * @param diagnosticMap maps paths (string) to diagnostic lists
+   */
+  private setDiagnosticMap(
+    diagnostic: Diagnostic,
+    path: string,
+    diagnosticMap: Map<string, Diagnostic[]>,
+  ): void {
+    const diagnostics: Diagnostic[] = diagnosticMap.get(path) || []
+    diagnostics.push(diagnostic)
+    diagnosticMap.set(path, diagnostics)
+  }
+
+  /**
+   * post all diagnostic lists in diagnosticMap to [PROBLEMS]
+   * @param diagnosticMap maps paths (string) to diagnostic lists
+   */
+  private postDiagnostics(diagnosticMap: Map<string, Diagnostic[]>): void {
+    diagnosticMap.forEach((diagnosticList, path) => {
+      const curDiagnosticCollection: vscode.DiagnosticCollection =
+        vscode.languages.createDiagnosticCollection()
+      curDiagnosticCollection.set(Uri.parse(path), diagnosticList)
+      this.diagnosticCollection.push(curDiagnosticCollection)
+    })
+    // when user edits file with diagnostics - clear related diagnostics
+    if (this.diagnosticCollection.length > 0) {
+      this.deleteOnEdit()
+    }
+  }
+
+  private getResourceError(str: string): ResourceError {
+    try {
+      const jsonContent: ResourceError = JSON.parse(str)
+      return jsonContent
+    } catch (e) {
+      console.error("Couldn't read the error logs. Please contact Certora team")
+      const resource_error: ResourceError = {
+        topics: [],
+      }
+      return resource_error
+    }
+  }
+
+  private async findFile(filepath: string): Promise<Uri[]> {
+    const ignoreFolderRegex =
+      '**/{.certora_config,.git,.last_confs,node_modules,certora-logs,conf,.github,images,cache}/**'
+    const found = await vscode.workspace.findFiles(
+      '**/' + filepath,
+      ignoreFolderRegex,
+      1,
+    )
+    return found
+  }
+
+  private async relativeToFullPath(uri: Uri): Promise<Uri> {
```

Comment:
There should be a library we can use for this (`path.resolve()`).
Working with paths is error-prone. You should take into account different file paths formats (e.g., Windows).
Of course, we'll test it on the different platforms before the deployment. Still, let's minimize potential errors.

Reply:
I changed `findFile` function to work with the path without having to process it...
The vscode api has a `Uri.joinPath` method that I'm using when I don't want to look recursively, but I think maybe in this case looking recursively may benefit us as the path we are getting from `resource_errors ` differ from message to message